apis/google/youtube/code/tool.gpt

```diff
@@ -0,0 +1,17 @@
+name: transcriber
+Description: Get a text transcript of a youtube video and clean it up.
+param: video_url: The url of a youtube video. (e.g. https://www.youtube.com/watch?v=VIDEO_ID or https://youtu.be/VIDEO_ID)
+tool: youtubeTranscriber, sys.write
+
+First, take the video_url parameter and pass it to the youtubeTranscriber tool.
+Process that transcript and add the missing punctuation. Clean up any misspellings or errors in the transcript.
+Do not truncate or reword any of the text, just add punctuation and correct mistakes.
+Write the corrected text to a file in the $GPTSCRIPT_WORKSPACE_DIR with a filename based on the VIDEO_ID.
```

Comment:
I think you need to add a context tool in order for this to know what $GPTSCRIPT_WORKSPACE_DIR actually is. Here is our pre-made one: https://github.com/gptscript-ai/context/tree/main/workspace

Reply:
hmm... true. i think this might have worked for me because i was using my current dir as the workspace. I've pushed a commit to add the workspace tool and tested that it wrote the file to a workspace outside my current dir